### DISCUSSION TITLE
Using a cluster of replicas in multiple regions with apps

### DISCUSSION THREAD

**Discussion by ku9nov** [cloudnative-pg/cloudnative-pg#3474] [General]
Hello, and thank you for this wonderful project. Currently, I have deployed a self-managed Kubernetes cluster in the USA, where I have configured an application and a Postgres cluster replica with WAL replication to S3.

Example:
```
app-pod-1
app-pod-2
# They interact with the Postgres cluster replicas' endpoints in the SAME cluster
indexer-dev-db-r
indexer-dev-db-ro
indexer-dev-db-rw
```
I can set up an additional replica in another cluster that operates in a different region and VPC (hereinafter "the second cluster"), and it seems to work as expected. However, I was surprised that when creating a cluster of replicas in the second cluster, it creates access points with the same names:
```
indexer-dev-db-r    ClusterIP   10.109.130.141   <none>        5432/TCP   4m25s
indexer-dev-db-ro   ClusterIP   10.102.12.65     <none>        5432/TCP   4m24s
indexer-dev-db-rw   ClusterIP   10.109.56.107    <none>        5432/TCP   4m24s
```
I mean, this replica should definitely not have a write function for now. 
**I want to achieve the following behavior:**
### When everything is normal
**First Cluster**

- The application works with the main database located in this cluster.

**Second Cluster**
- The application also works with the main database from the first cluster.

### When the first cluster dies
**Second Cluster**
- The replica becomes the main database.
- The application starts working with the database in the second cluster.
- After the recovery of the first cluster It starts working with the main database located in the second cluster.

So I have several questions:

1. How do I connect my app-pods in the second cluster to my main database in the first cluster? Do I need to use services or Istio? Or is there a solution in this project that I did not understand? Because having identical service names is a bit confusing.
2. In case the first cluster dies, will the second one automatically become the main one? Do I need to manage access to it through Istio?
3. After the recovery of the first cluster, will it automatically become a replica of the second cluster?

**Comment by gbartolini** [comment]
We will be addressing enhancements of the replica cluster feature in 2024, introducing support for planned switchover, additional replication slots and sync standby names (in case you have replica clusters close to each other).

However, coordination among different Kubernetes clusters is not in scope at the moment.

**Reply by ku9nov** [reply]
sounds good
thanks
